lib/src/main/java/com/squareup/cycler/RecyclerMutations.kt

```diff
@@ -1,24 +1,55 @@
 package com.squareup.cycler
 
+import android.graphics.Canvas
 import android.view.MotionEvent
 import android.view.View
+import android.view.View.MeasureSpec
+import androidx.annotation.VisibleForTesting
 import androidx.recyclerview.widget.ItemTouchHelper
 import androidx.recyclerview.widget.RecyclerView
 import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import com.squareup.cycler.Recycler.CreatorContext
+import com.squareup.cycler.Recycler.RowSpec
 import com.squareup.cycler.StandardRowSpec.Creator
+import com.squareup.cycler.SwipeDirection.END
+import com.squareup.cycler.SwipeDirection.LEFT
+import com.squareup.cycler.SwipeDirection.RIGHT
+import com.squareup.cycler.SwipeDirection.START
+import java.util.EnumSet
+import kotlin.math.abs
 
 fun <I : Any> Recycler.Config<I>.enableMutations(block: MutationExtensionSpec<I>.() -> Unit) {
   extension(MutationExtensionSpec<I>().apply(block))
 }
 
+/**
+ * Used in swipe related methods. They can be absolute or text-direction relative ([START]/[END]).
+ * The values used will depend on what is returned in [MutationExtensionSpec.canSwipe].
+ */
+enum class SwipeDirection(val absolute: Boolean, val platformValue: Int) {
+  START(false, ItemTouchHelper.START),
+  END(false, ItemTouchHelper.END),
+  LEFT(true, ItemTouchHelper.LEFT),
+  RIGHT(true, ItemTouchHelper.RIGHT);
+
+  companion object {
+    val NONE = EnumSet.noneOf(SwipeDirection::class.java)
+    val BOTH = EnumSet.of(START, END)
+    val BOTH_ABSOLUTE = EnumSet.of(LEFT, RIGHT)
+  }
+}
+
 class MutationExtensionSpec<T : Any> : ExtensionSpec<T> {
   override fun create(): Extension<T> = MutationExtension(this)
   internal var onMove: ((Int, Int) -> Unit)? = null
-  internal var onSwipeToRemove: ((T) -> Unit)? = null
-  internal var canSwipeToRemoveItem: ((T) -> Boolean) = { true }
+  internal var onSwiped: ((T, direction: SwipeDirection) -> Unit)? = null
+  internal var canSwipeItem: ((T) -> EnumSet<SwipeDirection>) = { SwipeDirection.NONE }
```

Comment:
`EnumSet` seems like an Implementation detail, this should probably just be `Set`.

Reply:
My bad. I know I had some issue with defining an agnostic set, but didn't realize that I should use it anyway for the API!